### PR TITLE
Lbaker/land phase2

### PR DESCRIPTION
**Pull request by lbakerchef**
### Tasks

- [x] fix dependency issue for observer_cli
- [x] move to mainline ibrowse
- [x] update rebar for mover
- [x] update mover
- [x] fix syntax error in rebar.config: "*" -> ".*"
- [x] update rebar.lock files 
need `src/oc_erchef/apps/data_collector/rebar.lock`
- [x] rebar.config homogenization
- [x] revert jiffy to previous version
errors
- [x] erlsom -> integer_long_string_probs2
- [x] mini-s3 -> master
found this in `bookshelf/stest/rebar.config`:
github.com/markan/mini_s3, branch ma/fix_for_V18
- [x] opscoderl_httpc -> master: 
- [x] ibrowse -> ma/revert_ipv6 (bugs in ipv6)
- [x] opscoderl_httpc -> branch master
- [x] ibrowse -> ma/revert_ipv6
- [x] erlsom -> integer_long_string_probs2
- [x] remove erroneously-added deps from bookshelf 
- [x] remove erroneously-added dep from bifrost
- [x] remove erroneously-added dep from oc_erchef
- [x] `tag` changed to `branch`
- [x] `https` changed to `git`
- [x] alphabetize _checkouts/oc_erchef/rebar.config
- [x] upgrade deps _checkouts/oc_erchef/rebar.config
- [x] jiffy -> 051a74338c089c39f09532188bc82cf1adedbdc8
- [ ] run mark's script for checkout SHAs
- [x] squash all commits
- [x] verify pipeline green
https://buildkite.com/chef/chef-chef-server-master-verify/builds/573
- [x] adhoc pipeline green
https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/528#fead40d1-81a0-4d5b-9915-2ba7be5ec291
- [x] local build
- [x] `chef-server-ctl reconfigure` passes
- [x] `chef-server-ctl test` passes
- [x] `chef-server-ctl psql <...>` passes

### Questions
- should `mini_s3 point` to `ma/ibrowse_uses_mainline`, or `master` (currently `master`) ?
- opscoderl_httpc (choose branch):
ma/ibrowse_uses_mainline
master <------ choose this
   (`./src/oc_erchef/apps/data_collector/rebar.config` uses `ma/ibrowse_uses_mainline`
   all other `rebar.config` use `master`)
 - ibrowse (choose repo + branch):
"https://github.com/opscode/ibrowse.git",        {tag, "v4.0.1.1"}
"https://github.com/chef/ibrowse.git",               {branch, "ma/revert_ipv6"}
"https://github.com/cmullaparthi/ibrowse.git",  {branch, "master"
- remove `rebar3` (added by Mark)?
YES
- revert `Makefile` changes (added by Mark)?
run eunit and DROP

### CODE REVIEW

File: src/oc_erchef/rebar.config [hunk 1]
```diff
@@ -60,17 +59,19 @@
...
         {opscoderl_httpc, ".*",
-         {git, "https://github.com/chef/opscoderl_httpc", {branch, "master"}}},
+         {git, "https://github.com/chef/opscoderl_httpc", {branch, "ma/ibrowse_uses_mainline"}}},
         {sync, ".*",
          {git, "https://github.com/rustyio/sync.git", {branch, "master"}}},
         {prometheus, ".*",
          {git, "https://github.com/deadtrickster/prometheus.erl", {tag, "v3.4.0"}}},
+        {eper, ".*",
+         {git, "https://github.com/massemanet/eper", {branch, "master"}}},
         {observer_cli, ".*",
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/rebar.config]: Any reason for changing the order here?

File: src/oc_erchef/rebar.config [hunk 2]
```diff
@@ -60,17 +59,19 @@
...
          {git, "https://github.com/zhongwencool/observer_cli", {branch, "master"}}}
        ]}.
 
-{cover_enabled, true}.
-{cover_print_enabled, true}.
+%{cover_enabled, true}.
+%{cover_print_enabled, true}.
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/rebar.config]: can delete the commented section if it is not wanted.

File: src/oc_erchef/Makefile [hunk 3]
```diff
@@ -21,7 +21,10 @@ DB_PASS = "erchef_test_user_password"
 include Makefile.db
 
 all: $(REBAR3) elvis
-	@$(REBAR3) do clean, compile, eunit, dialyzer
+	@$(REBAR3) do clean, compile
+	$(REBAR3) eunit --application chef_objects
+	$(REBAR3) eunit 
+	$(REBAR3) dialyzer
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/Makefile]: Can we use the original command?

**Comment by PrajaktaPurohit** [comment]
commit: 03b40d901c5e2e0263a090e1ab54912ad8f25456 adds the rebar3 exe. We should get rid of that as a part of this pr.

File: src/bookshelf/rebar.config [hunk 4]
```diff
@@ -8,38 +8,45 @@
...
 {deps, [
         {lager,".*",
-         {git,"https://github.com/basho/lager", {tag, "2.1.1"}}},
+         {git,"https://github.com/basho/lager.git", {tag, "2.1.1"}}},
         {meck, ".*",
          {git, "https://github.com/eproxus/meck.git", {tag, "0.8.4"}}},
         {chef_secrets, ".*",
-         {git, "https://github.com/chef/chef_secrets", {branch, "master"}}},
+         %% ma/bump_versions is temporary until branch is merged.
+         {git, "https://github.com/chef/chef_secrets.git", {branch, "ma/bump_versions"}}},
```
Comment (PrajaktaPurohit) [inline src/bookshelf/rebar.config]: ensure we are on master for this.

File: src/bookshelf/rebar.config [hunk 5]
```diff
@@ -8,38 +8,45 @@
...
         {iso8601, ".*",
-         {git, "https://github.com/erlsci/iso8601", {tag, "1.2.3"}}},
+         {git, "https://github.com/erlsci/iso8601.git", {tag, "1.2.3"}}},
         {opscoderl_wm, ".*",
          {git, "https://github.com/chef/opscoderl_wm.git", {branch,"master"}}},
         {erlsom, ".*",
-         {git, "https://github.com/chef/erlsom",
+         {git, "https://github.com/chef/erlsom.git",
           {branch, "integer_long_string_probs"}}},
```
Comment (PrajaktaPurohit) [inline src/bookshelf/rebar.config]: can we get on the master for this? looks like there is also a integer_long_string_probs2 branch that has some more updates.

File: src/bookshelf/rebar.config [hunk 6]
```diff
@@ -2,45 +2,103 @@
...
+    {lager, ".*",
+        {git, "https://github.com/basho/lager.git",                     {branch, "master"}}},
+%   {bcrypt, ".*",
+%       {git, "https://github.com/smarkets/erlang-bcrypt.git",          {branch, "master"}}},
+    {cf, "",
+        {git, "https://github.com/project-fifo/cf.git",                 {branch, "master"}}},
+%   {chef_authn, ".*",
+%       {git, "https://github.com/chef/chef_authn.git",                 {branch, "master"}}},
+%   {chef_reindex, ".*",
+%       {git, "https://github.com/opscode/chef_reindex.git",            {branch, "master"}}},
```
Comment (PrajaktaPurohit) [inline src/bookshelf/rebar.config]: We should get rid of the commented dependencies if we do not need them.

File: src/oc_erchef/apps/data_collector/rebar.config [hunk 7]
```diff
@@ -15,9 +15,10 @@
 ]}.
 
 {deps, [
-    {lager, ".*", {git, "https://github.com/basho/lager", {tag, "2.1.1"}}},
-    {opscoderl_httpc, ".*", {git, "https://github.com/chef/opscoderl_httpc", {branch, "master"}}},
-    {pooler, ".*", {git,"git://github.com/seth/pooler.git", {branch,"master"}}}
+    {lager, ".*", {git, "https://github.com/basho/lager.git", {tag, "2.1.1"}}},
+    {opscoderl_httpc, ".*",
+     {git, "https://github.com/chef/opscoderl_httpc.git", {branch, "ma/ibrowse_uses_mainline"}}},
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/apps/data_collector/rebar.config]: any reason to use ma/ibrowse_uses_mainline? looks like we have merged it in the opscoderl_httpc repo.

File: src/oc_erchef/rebar.config [hunk 8]
```diff
@@ -2,97 +2,123 @@
 %% -*- tab-width: 4;erlang-indent-level: 4;indent-tabs-mode: nil -*-
 %% ex: ts=4 sw=4 ft=erlang et
 
+
+
+
+
+
+
+
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/rebar.config]: can delete the empty newlines

File: src/bookshelf/rebar.config (hunk 4 above)
Comment (PrajaktaPurohit) [reply src/bookshelf/rebar.config]: yep - updated in a later commit.

File: src/bookshelf/rebar.config (hunk 6 above)
Comment (PrajaktaPurohit) [reply src/bookshelf/rebar.config]: Applicable in other places that have commented dependencies.

**Comment by PrajaktaPurohit** [comment]
the data_collector/rebar.lock file needs to be added in.

File: src/oc_erchef/rebar.config (hunk 1 above)
Comment (lbakerchef) [reply src/oc_erchef/rebar.config]: I forgot why.  However, all the `rebar.config`s were redone in alphabetical order in a later commit.

File: src/oc_erchef/rebar.config (hunk 2 above)
Comment (lbakerchef) [reply src/oc_erchef/rebar.config]: commenting those out was a mistake.  good catch!

File: src/bookshelf/rebar.config (hunk 6 above)
Comment (lbakerchef) [reply src/bookshelf/rebar.config]: i'll show you why i have the comments there when we can zoom.

**Comment by markan** [comment]
Questions from above:

in general, we should see if master works now; otherwise we probably have to do work around ibrowse_uses_mainline (pretty old now) to see what's required to revive it.

- should `mini_s3 point` to `ma/ibrowse_uses_mainline`, or `master` (currently `master`) ?

Looks like mini_s3 rebar.conf points to a branch of master because of the ipv6 issues; we need to dive into that more deeply.

- opscoderl_httpc (choose branch):
ma/ibrowse_uses_mainline
master
   (`./src/oc_erchef/apps/data_collector/rebar.config` uses `ma/ibrowse_uses_mainline`
   all other `rebar.config` use `master`)



 - ibrowse (choose repo + branch):
"https://github.com/opscode/ibrowse.git",        {tag, "v4.0.1.1"}
"https://github.com/chef/ibrowse.git",               {branch, "ma/revert_ipv6"}

chef and opscode are the same repo, opscode just forwards to chef.

"https://github.com/cmullaparthi/ibrowse.git",  {branch, "master"
- remove `rebar3` (added by Mark)?

I'd remove that and update rebar3 separately, maybe even first.

- revert `Makefile` changes (added by Mark)?
Yes, drop the makefile changes, and revisit it later.

File: src/oc_erchef/apps/data_collector/rebar.config (hunk 7 above)
Comment (lbakerchef) [reply src/oc_erchef/apps/data_collector/rebar.config]: Fixed.

File: src/bookshelf/rebar.config (hunk 5 above)
Comment (lbakerchef) [reply src/bookshelf/rebar.config]: trying `integer_long_string_probs2`

**Comment by PrajaktaPurohit** [comment]
Passing adhoc build: https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/596#3a12138e-114a-40ea-b30d-89a53d1416f8

**Comment by PrajaktaPurohit** [comment]
Just a quick question - so we want to include the rebar.lock for src/chef-mover/_checkouts/oc_erchef? Other than that this looks good to merge!

File: src/chef-mover/rebar.config [hunk 9]
```diff
@@ -1,62 +1,74 @@
...
+    {edown, ".*",
+        {git, "https://github.com/uwiger/edown.git",                    {branch, "master"}}},
+    {ej, ".*",
+        {git, "https://github.com/seth/ej.git",                         {branch, "master"}}},
+    {eper, ".*",
+        {git, "https://github.com/massemanet/eper.git",                 {branch, "master"}}},
+    {epgsql,".*",
+        {git, "https://github.com/chef/epgsql-1.git",                   {branch, "master"}}},
+    {eredis, ".*",
+        {git, "https://github.com/wooga/eredis.git",                    {tag, "master"}}},
```
Comment (markan) [inline src/chef-mover/rebar.config]: Also {tag, "master"} => {branch, "master"}

File: src/bookshelf/rebar.config [hunk 10]
```diff
@@ -2,45 +2,51 @@
...
+    {chef_secrets, ".*",
+        {git, "https://github.com/chef/chef_secrets.git",               {branch, "master"}}},
+    {ej, ".*",
+        {git, "https://github.com/seth/ej.git",                         {branch, "master"}}},
+    {envy, ".*",
+        {git, "https://github.com/markan/envy.git",                     {branch, "master"}}},
+    {eper, ".*",
+        {git, "https://github.com/massemanet/eper.git",                 {branch, "master"}}},
+    {eredis, ".*",
+        {git, "https://github.com/wooga/eredis.git",                    {tag, "master"}}},
```
[the remark this answers is not in the document]
Comment (lbakerchef) [reply src/bookshelf/rebar.config]: good catch

File: src/chef-mover/rebar.config (hunk 9 above)
Comment (lbakerchef) [reply src/chef-mover/rebar.config]: was {tag, "master"} in master branch.  now changed to {branch, master}.

File: src/bookshelf/rebar.config (hunk 10 above)
[the remark this answers is not in the document]
Comment (lbakerchef) [reply src/bookshelf/rebar.config]: good catch, as eredis isn't even supposed to be there.

fixed.

**Comment by lbakerchef** [comment]
> Looks very good generally. A few inline notes.
> One ask: can we document (maybe in a comment) why we are using branches and specific SHAs or versions in the rebar.conf?
> For example we pin to a particular SHA of davisp/jiffy; why is that? We may know that offhand now, but likely will forget quickly.

Good idea.  In general I don't know the reasons for any specific SHAs or versions, I merely tried to upgrade masters to latest masters, but in a few cases I also updated some tags to masters.  Will look through your earlier commits to see if I can find any info there.

**Comment by lbakerchef** [comment]
> If you have a chance, please check that all instances of the same package is pinned to the same SHA.

Will do.  I also plan to run your script.

Got Jiffy bisected up from the older version.  Also, updated deps in _checkouts.

**Comment by lbakerchef** [comment]
https://buildkite.com/chef/chef-chef-server-master-verify/builds/649#c6ff9921-903f-45a8-ad76-b3f58a01a14a

File: src/oc_erchef/apps/data_collector/rebar.lock [hunk 11]
```diff
@@ -1,20 +1,23 @@
...
-  1},
+[
+ %% these apps are not present in the rebar.conf so
+ %% commenting out for now to see what happens.
+%{<<"goldrush">>,
+% {git,"git://github.com/DeadZen/goldrush.git",
+%      {ref,"57e71f2789632a085f0963e1244644fdcb1941b9"}},
+% 1},
+%{<<"ibrowse">>,
+% {git,"git://github.com/chef/ibrowse.git",
```
Comment (tas50) [inline src/oc_erchef/apps/data_collector/rebar.lock]: If we find out eventually we're not using this then we should move that repo over to the chef-boneyard org.

Comment (lbakerchef) [reply src/oc_erchef/apps/data_collector/rebar.lock]: Sure, or just delete entirely.

Comment (tas50) [reply src/oc_erchef/apps/data_collector/rebar.lock]: Usually we just move them just in case

**Comment by PrajaktaPurohit** [comment]
closes #1751 https://github.com/chef/chef-server/issues/1780

**Pull request merged**